src/test/java/com/amazon/opendistroforelasticsearch/knn/index/KNNESIT.java

```diff
@@ -325,4 +329,36 @@ public void testInvalidQueryHnswAlgoParams() {
                 createKnnIndexMapping(FIELD_NAME, 2)));
         assertThat(ex.getMessage(), containsString("Failed to parse value [-1] for setting [index.knn.algo_param.ef_search]"));
     }
+
+    public void testExistsQuery() throws IOException {
```

Comment:
How about we declare 2 different knn fields(k1, k2) in the mappings with below scenario

Have 1 document with both the k1,k2.
Have 3 docs with only k1
Have 2 docs with only k2
Have 1 doc with neither k1 nor k2

With exists query for k1, assert hits=4
With exists query for k2, assert hits=3

Reply:
Makes sense. Updated the test cases.